enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -131,20 +136,49 @@ const InteractiveDashboardInner = ({
 };
 
 export const InteractiveDashboard = renderOnlyInSdkProvider(
-  ({ dashboardId, ...rest }: InteractiveDashboardProps) => {
-    const { id, isLoading } = useValidatedEntityId({
+  ({ dashboardId: initialDashboardId, ...rest }: InteractiveDashboardProps) => {
+    const { id: resolvedDashboardId, isLoading } = useValidatedEntityId({
       type: "dashboard",
-      id: dashboardId,
+      id: initialDashboardId,
     });
 
+    const errorPage = useSdkSelector(getErrorPage);
+    const dispatch = useSdkDispatch();
+    useEffect(() => {
+      if (resolvedDashboardId) {
+        dispatch(setErrorPage(null));
+      }
+    }, [dispatch, resolvedDashboardId]);
+
+    const { style, className } = rest;
     if (isLoading) {
-      return <SdkLoader />;
+      return (
+        <StyledPublicComponentWrapper className={className} style={style}>
```

Comment:
Fix the issue where the loading state isn't properly themed. This was outside the scope, but I added it for completeness.